core/src/main/java/com/linecorp/armeria/server/management/AppInfoService.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License
+ */
+
+package com.linecorp.armeria.server.management;
+
+import com.google.common.collect.ImmutableMap;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.util.Version;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * An {@link HttpService} that provides additional information about configured server.
+ * This functionality is can be used by binding {@link ManagementService}
+ * It provides information about not only deployed application information, which can be specified by user,
+ * but also one about Armeria artifact itself.
+ */
+public enum AppInfoService implements HttpService {
+
+    INSTANCE;
+
+    @Nullable
+    private AppInfo appInfo;
+
+    void setAppInfo(@Nullable AppInfo appInfo) {
+        this.appInfo = appInfo;
+    }
+
+    @Override
+    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+        final Version versionInfo = Version.get("armeria", Server.class.getClassLoader());
+
+        final ImmutableMap<String, ImmutableMap<String, String>> appInfoMap;
+        if (appInfo != null) {
+            appInfoMap = ImmutableMap.of(
+                    "app", ImmutableMap.of(
+                            "version", appInfo.version,
+                            "name", appInfo.name,
+                            "description", appInfo.description
```

Comment:
Should we annotate getters of `AppInfo` with `@JsonProperty` to directly serialize the value? For example:
https://github.com/line/armeria/blob/e1d137738a8e4f3f7a7943e69f3f314ab772e338/core/src/main/java/com/linecorp/armeria/common/util/Version.java#L263-L264